enhancements/cluster-logging/cluster-logging-kibana-multitenancy.md

```diff
@@ -0,0 +1,188 @@
+  - "@richm"
+creation-date: 2019-12-10
+last-updated: 2019-12-10
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# cluster-logging-kibana-multitenancy
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Migration plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of updating the Kibana Multitenancy pattern is to no longer rely on the
+[openshift-elasticsearch-plugin](https://github.com/fabric8io/openshift-elasticsearch-plugin/) to
+provide multitenancy for Kibana. In conjunction with the [elasticsearch-proxy](https://github.com/ewolinetz/enhancements/blob/master/enhancements/cluster-logging/cluster-logging-elasticsearch-proxy.md) this would greatly simplify the openshift-elasticsearch-plugin and potentially make it
+obsolete.
+
+## Motivation
+
+Currently the openshift-elasticsearch-plugin intercepts calls from Kibana, rewrites them
+based on the username to match up with an user-specific index, then rewrites the response so that Kibana does not detect that things have been rewritten. This currently requires that we are able to
+keep up with any changes to the Kibana -> Elasticsearch API for intercepting and updating
+the calls, this instead would be handled by Kibana before sending a request to Elasticsearch.
+
+### Goals
+The specific goals of this proposal are:
+
+* Provide a means to link OCP users to Kibana tenants and provide Elasticsearch log access
+based on their OCP access.
+
+We will be successful when:
+
+* An OCP user is able to log into Kibana and have a user specific tenant provided for them
+* Their tenant has the appropriate index patterns assigned to them
+* They should be able to share visualizations between accounts
+
+### Non-Goals
+
+* This is to configure Kibana such that users have their user tenants created for them and the appropriate
+index patterns are made available. Role access is separate from this implementation and will only be tying
+an index pattern to a specific user's role.
+
+## Proposal
+
+The OpenDistro Multitenancy plugin seeks to simplify our current process of catching requests from Kibana before they are processed by elasticsearch where we then rewrite the index to uniquely match an user. The tenant patterns that are matched to
+an user are defined within the user roles that are seeded into Elasticsearch. This functionality is to be provided by the Elasticsearch Proxy.
+
+### User Stories
+
+#### As an OKD admin, I want to be able to view all logs that I have access to
+
+This is the current state, admin users should still be able to view all the cluster logs with the provided index patterns
+
+#### As an OKD user, I want to be able to view all logs that I have access to
```

Comment:
`As an OKD user, I want to be able to view all container logs from the namespaces to which I have access`